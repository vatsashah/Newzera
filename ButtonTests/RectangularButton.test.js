// Importing required library
import React from 'react'
import { shallow, mount } from 'enzyme'

// Importing component to be tested
import RectangleButton from '../../../Components/Buttons/defaultButton'

describe('RectangleButton should render without any error', () => {
  it('RectangleButton should render correctly', () => {
    const component = shallow(<RectangleButton />)
    expect(component).toMatchSnapshot()
  })
  it('RectangleButton should be disabled', () => {
    const wrapper = mount(<RectangleButton disabled />).props()
    expect(wrapper.disabled).toBe(true)
  })
  it('RectangleButton should be bordered', () => {
    const wrapper = mount(<RectangleButton bordered />).props()
    expect(wrapper.bordered).toBe(true)
  })
  it('RectangleButton should be filled', () => {
    const wrapper = mount(<RectangleButton filled />).props()
    expect(wrapper.filled).toBe(true)
  })
})
