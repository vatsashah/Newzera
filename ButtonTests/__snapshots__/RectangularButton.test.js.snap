// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RectangleButton should render without any error RectangleButton should render correctly 1`] = `
<div
  className="RectangleDefaultBackground"
  style={
    Object {
      "borderRadius": "0vh",
      "height": "4.62962962962963vh",
      "width": "15.625vw",
    }
  }
>
  <p
    className="RectangleDefaultText"
    style={
      Object {
        "fontSize": "1.6666666666666667vh",
        "height": "1.6666666666666667vw",
        "top": "1.4814814814814814vh",
        "width": "15.625vw",
      }
    }
  >
    Button
  </p>
</div>
`;
